src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,46 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static nature() {
+    this.alive = this.alive.filter((dead) =>
+      dead.health !== 0);
+  }
+  constructor(name, health = 100) {
+    this.health = health;
+    this.name = name;
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, health = 100, toHide = false) {
+    super(name, health);
+    this.hidden = toHide;
+
+    Animal.alive.push(this);
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
+  bite(mammal) {
+    if (mammal.hidden === false) {
+      mammal.health -= 50;
+
+      if (mammal.health === 0) {
+        Animal.nature();
+      }
+    }
```

Comment:
[CODE LOGIC] - The 'bite' method should check if the 'mammal' is an instance of Herbivore and not currently hiding. Currently, it only checks if 'mammal.hidden' is false, which will not work correctly if a Carnivore tries to bite another Carnivore. Additionally, the method should not decrease health if the mammal is hiding. Consider adding checks for instance type and whether the mammal is hiding.